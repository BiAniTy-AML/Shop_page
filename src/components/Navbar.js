import { Link } from "react-router-dom";

const Navbar = () => {
    return (
        <nav>
            <h1 className="logo">Logo</h1>

            <ul>
                <li>
                    <Link to={"/"}>Home</Link>
                </li>
                <li>
                    <Link to={"/shop"}>Shop</Link>
                </li>
                <li>Cart</li>
            </ul>
        </nav>
    );
};

export default Navbar;
