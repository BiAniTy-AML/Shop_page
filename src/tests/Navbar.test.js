import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import Navbar from "../components/Navbar";
import { MemoryRouter } from "react-router-dom";

describe("Testing navbar component", () => {
    it("renders the navbar element", () => {
        render(
            <MemoryRouter>
                <Navbar />
            </MemoryRouter>
        );

        const navbar = screen.getByRole("navigation");

        expect(navbar).toBeInTheDocument();
        expect(navbar).toMatchSnapshot();
    });

    it("renders the navbar li's correctly", () => {
        render(
            <MemoryRouter>
                <Navbar />
            </MemoryRouter>
        );

        const pages_links = screen.getAllByRole("listitem");

        expect(pages_links.length).toBe(3);
        expect(pages_links[0].textContent).toMatch("Home");
        expect(pages_links[1].textContent).toMatch("Shop");
        expect(pages_links[2].textContent).toMatch("Cart");
    });

    it("renders the logo placeholder correctly", () => {
        render(
            <MemoryRouter>
                <Navbar />
            </MemoryRouter>
        );

        const logo_placeholder = screen.getByRole("heading");

        expect(logo_placeholder.textContent).toMatch("Logo");
    });
});
