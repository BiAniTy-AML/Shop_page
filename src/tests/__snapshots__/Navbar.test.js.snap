// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing navbar component renders the navbar element 1`] = `
<nav>
  <h1
    class="logo"
  >
    Logo
  </h1>
  <ul>
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/shop"
      >
        Shop
      </a>
    </li>
    <li>
      Cart
    </li>
  </ul>
</nav>
`;
